### ISSUE TITLE
`use-implicit-booleaness-not-comparison` should not be offered in N-way comparisons

### ISSUE THREAD

**Issue by stdedos** [pylint-dev/pylint#10065]
### Bug description

```py
assert a == b == ''
```

### Command used

pylint a.py

### Pylint output

C1804: "a == b == ''" can be simplified to "not b", if it is striclty a string, as an empty string is falsey (use-implicit-booleaness-not-comparison-to-string)

### Expected behavior

Nothing _or an acceptable fix_

### Pylint version

pylint 3.2.2
astroid 3.2.2
Python 3.9.20 (main, Sep  7 2024, 18:35:26) 
[GCC 9.4.0]

------------

pylint 3.3.1
astroid 3.3.5
Python 3.9.20 (main, Sep  7 2024, 18:35:26) 
[GCC 9.4.0]
